tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -0,0 +1,429 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+
+#include <nodes/bitmapset.h>
+#include <utils/builtins.h>
+#include <utils/date.h>
+#include <utils/timestamp.h>
+
+#include "nodes/decompress_chunk/compressed_batch.h"
+
+#include "compression/arrow_c_data_interface.h"
+#include "compression/compression.h"
+#include "debug_assert.h"
+#include "guc.h"
+#include "nodes/decompress_chunk/exec.h"
+
+void
+compressed_batch_set_compressed_tuple(DecompressChunkState *chunk_state,
+									  DecompressBatchState *batch_state, TupleTableSlot *subslot)
+{
+	Assert(TupIsNull(batch_state->decompressed_scan_slot));
+
+	/*
+	 * The batch states are initialized on demand, because creating the memory
+	 * context and the tuple table slots is expensive.
+	 */
+	if (batch_state->per_batch_context == NULL)
+	{
+		batch_state->per_batch_context =
+			AllocSetContextCreate(CurrentMemoryContext,
+								  "DecompressChunk per_batch",
+								  0,
+								  chunk_state->batch_memory_context_bytes,
```

Comment:
Could you comment on why the custom values are used for this memory context instead of `ALLOCSET_DEFAULT_SIZES`?

Reply:
Added.